contracts/external/cw-payroll/src/contract.rs

```diff
@@ -0,0 +1,1462 @@
+        let msg = cw20.call(Cw20ExecuteMsg::Transfer {
+            recipient: recepient,
+            amount: amount.unwrap_or(cw20_balance.amount),
+        })?;
+        return Ok(msg);
+    }
+    Err(ContractError::CouldNotCreateBankMessage {})
+}
+pub fn execute_resume_stream(
+    env: Env,
+    deps: DepsMut,
+    info: MessageInfo,
+    id: StreamId,
+) -> ContractResult {
+    let resume_stream_local =
+        |stream_id: StreamId, storage: &mut dyn Storage| -> Result<Stream, ContractError> {
+            let mut stream = STREAMS
+                .may_load(storage, stream_id)?
+                .ok_or(ContractError::StreamNotFound { stream_id: id })?;
+            if stream.admin != info.sender {
+                return Err(ContractError::Unauthorized {});
+            }
+            if !stream.paused {
+                return Err(ContractError::StreamNotPaused {});
+            }
+            stream.paused_duration = stream.calc_pause_duration(env.block.time);
+            stream.paused = false;
+            stream.paused_time = None;
+            save_stream(storage, id, &stream).unwrap();
+            Ok(stream)
+        };
+
+    let stream = resume_stream_local(id, deps.storage)?;
+    if let Some(link_id) = stream.link_id {
+        resume_stream_local(link_id, deps.storage).unwrap();
+    }
+    let (_, rate_per_second) = stream.calc_distribution_rate(env.block.time);
+    let response = Response::new()
+        .add_attribute("method", "resume_stream")
+        .add_attribute("stream_id", id.to_string())
+        .add_attribute("admin", info.sender)
+        .add_attribute("rate_per_second", rate_per_second)
+        .add_attribute("resume_time", env.block.time.to_string())
+        .add_attribute(
+            "paused_duration",
+            stream.paused_duration.unwrap().to_string(),
+        )
+        .add_attribute("resume_time", env.block.time.to_string());
+
+    Ok(response)
+}
+pub fn execute_create_stream(env: Env, deps: DepsMut, params: StreamParams) -> ContractResult {
+    let StreamParams {
+        admin,
+        recipient,
+        balance,
+        start_time,
+        end_time,
+        title,
+        description,
+        is_detachable,
+    } = params;
+
+    let admin = deps.api.addr_validate(&admin)?;
+    let recipient = deps.api.addr_validate(&recipient)?;
+
+    if start_time > end_time {
+        return Err(ContractError::InvalidStartTime {});
+    }
+
+    let block_time = env.block.time.seconds();
+
+    if end_time <= block_time {
+        return Err(ContractError::InvalidEndTime {});
+    }
+
+    let duration: u128 = (end_time - start_time).into();
+    let balance_amount = balance.amount();
+    let refund: u128 = balance_amount
+        .checked_rem(duration)
+        .ok_or(GenericError::IntegerOverflow {})
+        .unwrap();
+
+    if balance_amount < duration {
+        return Err(ContractError::AmountLessThanDuration {});
+    }
```

Comment:
why do we need these restrictions on the amount being vested?

Reply:
Agree, we don't need this one